xtable-core/src/main/java/org/apache/xtable/delta/DeltaConversionSource.java

```diff
@@ -151,7 +153,7 @@ public TableChange getTableChangeForCommit(Long versionNumber) {
     // entry which is replaced by a new entry, AddFile with delete vector information. Since the
     // same data file is removed and added, we need to remove it from the added and removed file
     // maps which are used to track actual added and removed data files.
-    for (String deletionVector : deletionVectors) {
+    for (String deletionVector : deletionVectors.keySet()) {
```

Comment:
nitpick: the name `deletionVector` is no longer representative of the actual string. Something like `dataFileForDeletionVector` would be more clear